src/app/common/header/header-search.directive.js

```diff
@@ -0,0 +1,26 @@
+import angular from 'angular';
+
+export class HeaderSearch {
+  constructor() {
+    'ngInject';
+    this.restrict = 'A';
+    this.scope = {
+      fetchSearch: '&'
+    };
+  }
+  link($scope, $element) {
+    const limit = 4;
+    const handler = ($event) => {
```

Comment:
Probably playing some with ng-model-options we can achieve the same without having to do this one.
Using `getterSetter ` and binding a method instead of value and using `ng-change` to trigger that event